uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -259,75 +275,65 @@ INPUTS:
   - $subjectIDsref: subjectIDs hashref
   - $dbhr         : database handle reference
 
-RETURNS: the determined objective, or 0
+RETURNS: the determined objective, or undef
 
 =cut
 
 sub getObjective
 {
-    my ($subjectIDsref, $dbhr) = @_;
+    my ($subjectIDsref, $dbh) = @_;
     my @results = ();
     my $objective = 0;
     my %subjectIDs = %$subjectIDsref;
-    if($subjectIDs{'visitLabel'} =~ /PHA/i or $subjectIDs{'visitLabel'} =~ /TEST/i) {
-	return 0;
-    }
-
-    my $query = "SELECT SubprojectID FROM session WHERE CandID='$subjectIDs{'CandID'}' AND Visit_label='$subjectIDs{'visitLabel'}' AND Active='Y' ORDER BY ID DESC LIMIT 1";
-    my $sth = $${dbhr}->prepare($query) or die "Can't prepare $query: ".$${dbhr}->errstr."\n";
 
-    $sth->execute();
-
-    if($sth->rows > 0) {
-        @results = $sth->fetchrow_array();
-    }
+    my $query = "SELECT SubprojectID "
+              . "FROM session "
+              . "WHERE CandID = ? "
+              . "AND Visit_label = ? "
+              . "AND Active='Y' "
+              . "ORDER BY ID DESC LIMIT 1";
+    my $sth = $dbh->prepare($query) or die "Can't prepare $query: " . $dbh->errstr."\n";
 
-    $objective = $results[0] if $results[0];
-
-    unless($objective>0) {
-        # there probably isn't a valid row for this visit...
-        $query = "SELECT SubprojectID FROM session WHERE CandID='$subjectIDs{'CandID'}' AND Active='Y' ORDER BY ID DESC LIMIT 1";
-        $sth = $${dbhr}->prepare($query);
-        $sth->execute();
-
-        @results = $sth->fetchrow_array();
-
-        $objective = $results[0] if $results[0];
-    }
-    return $objective;
+    $sth->execute($subjectIDs{'CandID'}, $subjectIDs{'visitLabel'});
 
+    return undef if $sth->rows == 0;
+       
+    @results = $sth->fetchrow_array();
```

Comment:
since will return both SubprojectID and ProjectID, will probably be easier to grep a hash.

Also, this function is missing the return statement ;)

Reply:
Arghhh.....!!!!! Corrected.

Reply:
hihihi ;)